mlflow/pipelines/steps/train.py

```diff
@@ -124,135 +124,181 @@ def _run(self, output_directory):
-                if "values" in param_details:
-                    search_space[param_name] = hp.choice(param_name, **param_details)
-                elif "distribution" in param_details:
-                    hp_tuning_fn = getattr(hp, param_details["distribution"])
-                    param_details_to_pass = param_details.copy()
-                    param_details_to_pass.pop("distribution")
-                    search_space[param_name] = hp_tuning_fn(param_name, **param_details_to_pass)
-                else:
+        mlflow.autolog(log_models=False)
+        with mlflow.start_run(tags=tags) as run:
+            estimator_hardcoded_params = self.step_config["estimator_params"]
+            if self.step_config["tuning_enabled"]:
+                # gate all HP tuning code within this condition
+                tuning_params = self.step_config["tuning"]
+                # import hyperopt or throw error
+                try:
+                    from hyperopt import hp, fmin
+                except ModuleNotFoundError:
                     raise MlflowException(
-                        f"Parameter {param_name} must contain either a list of 'values' or a "
-                        f"'distribution' following hyperopt parameter expressions",
-                        error_code=INVALID_PARAMETER_VALUE,
+                        "Hyperopt not installed and is required if tuning is enabled",
+                        error_code=BAD_REQUEST,
                     )
-            # minimize
-            algorithm = tuning_params["algorithm"]  # pylint: disable=unused-variable
-            max_trials = tuning_params["max_trials"]  # pylint: disable=unused-variable
-            best = fmin(  # pylint: disable=unused-variable
-                objective, search_space, max_evals=max_trials
-            )
 
-        else:
-            estimator = estimator_fn()
-            mlflow.autolog(log_models=False)
-
-            with mlflow.start_run(tags=tags) as run:
-                estimator.fit(X_train, y_train)
-
-                if hasattr(estimator, "best_score_"):
-                    mlflow.log_metric("best_cv_score", estimator.best_score_)
-                if hasattr(estimator, "best_params_"):
-                    mlflow.log_params(estimator.best_params_)
-
-                code_paths = [os.path.join(self.pipeline_root, "steps")]
-                estimator_schema = infer_signature(X_train, estimator.predict(X_train.copy()))
-                logged_estimator = mlflow.sklearn.log_model(
-                    estimator,
-                    f"{self.name}/estimator",
-                    signature=estimator_schema,
-                    code_paths=code_paths,
+                # wrap training in objective fn
+                def objective(hyperparameter_args):
+                    # log as a child run
+                    with mlflow.start_run(tags=tags, nested=True):
+                        # create unfitted estimator from yaml
```

Comment:
Can we remove these comments unless the code that follows them is particularly hard to read / unintuitive?